src/js/utils/media-downloader.js

```diff
@@ -0,0 +1,62 @@
+
+var JSZip = require("JSzip")
+
+zip = new JSZip();
```

Comment:
The only thing I'd change in the commits so far.

I don't think you need this global variable (also could be abused if user spams download button).

You could make line 32, declare var zip and then just pass as a parameter to addUrlToZip.

Let me know what you think.

Reply:
Yep makes sense to make it non global